common/src/main/java/com/wynntils/core/webapi/profiles/MapProfile.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.webapi.profiles;
+
+import com.wynntils.mc.utils.McUtils;
+import net.minecraft.client.renderer.texture.DynamicTexture;
+import net.minecraft.resources.ResourceLocation;
+
+public class MapProfile {
+
+    DynamicTexture texture;
+    ResourceLocation mapResource;
+
+    int x1;
+    int z1;
+    int x2;
+    int z2;
+
+    int textureWidth;
+    int textureHeight;
+
+    public MapProfile(DynamicTexture texture, int x1, int z1, int x2, int z2, int textureWidth, int textureHeight) {
+        this.texture = texture;
+        this.x1 = x1;
+        this.z1 = z1;
+        this.x2 = x2;
+        this.z2 = z2;
+        this.textureWidth = textureWidth;
+        this.textureHeight = textureHeight;
+
+        // Remove this if we ever have non 1 to 1 maps
+        assert x2 - x1 == textureWidth;
+        assert z2 - z1 == textureHeight;
+    }
+
+    public float getTextureXPosition(double posX) {
+        return (float) (posX - x1);
+    }
+
+    public float getTextureZPosition(double posZ) {
+        return (float) (posZ - z2);
+    }
+
+    public int getWorldXPosition(double textureX) {
+        return (int) Math.round(textureX + x1);
+    }
+
+    public int getWorldZPosition(double textureY) {
+        return (int) Math.round(textureY + z1);
+    }
+
+    public ResourceLocation resource() {
+        if (mapResource == null) {
```

Comment:
Why is this not done in the constructor?

Reply:
I said more about this in the discord, but `TextureManager` is null when the mod initalizer for fabric is called, thus creating it in the constructor leads to a crash or an unregistered resource location.

Reply:
Does the crash happen with the ResourceLocation constructor, or the `register` call? If it is the latter, the ResourceLocation should still be created in the constructor.

Reply:
The `register` call. If the ResourceLocation is created in the constructor, then how can registration of texture manager be done later?

Reply:
There are many ways that could be done. Flag it with a boolean, for instance. The thing that bothers me with this code is actually how you hard-code string literals in the constructor. These should have been gotten from the constructor.

Reply:
This should be resolved.